game-core/src/test/java/org/triplea/lobby/common/GameDescriptionTest.java

```diff
@@ -3,46 +3,49 @@
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.is;
 
-import java.util.Optional;
+import java.util.Arrays;
+import java.util.Collection;
 
-import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.triplea.game.server.HeadlessGameServer;
 
 import games.strategy.engine.lobby.server.GameDescription;
 
 final class GameDescriptionTest {
-  @Nested
-  final class GetBotSupportEmailTest {
-    @Test
-    void shouldReturnBotSupportEmailWhenBotSupportEmailIsNotEmpty() {
-      final String botSupportEmail = "bot@me.com";
-      final GameDescription gameDescription = GameDescription.builder().botSupportEmail(botSupportEmail).build();
-
-      assertThat(gameDescription.getBotSupportEmail(), is(Optional.of(botSupportEmail)));
-    }
-
-    @Test
-    void shouldReturnEmptyWhenBotSupportEmailIsEmpty() {
-      final GameDescription gameDescription = GameDescription.builder().botSupportEmail("").build();
-
-      assertThat(gameDescription.getBotSupportEmail(), is(Optional.empty()));
-    }
+  @Test
+  void isBotShouldReturnTrueWhenCommentAndNamePrefixMatch() {
+    final GameDescription gameDescription = GameDescription.builder()
+        .comment(HeadlessGameServer.BOT_GAME_HOST_COMMENT)
+        .hostName(HeadlessGameServer.BOT_GAME_HOST_NAME_PREFIX)
+        .build();
+
+    assertThat(gameDescription.isBot(), is(true));
   }
 
-  @Nested
-  final class IsBotTest {
```

Comment:
FYI, nesting tests like this has the side effect of de-triggering the Eclipse static analyzer from wanting to say the test methods can be static (which they can't due to a JUnit requirement); that seemed preferable to adding a `@SuppressWarnings("static-method")` in such cases.  It also has the benefit of not repeating the name of the method under test in each test case.

Reply:
I should have called this one out:
- `@MethodSource` methods must be static
- nested classes may not contain nested methods

That was the motivation for removing the nesting. On a side note, IDEA declares the nested test classes as 'emtpy-classes' and we likely should suppress that warning. I agree to the benefit of not repeating test names.

Reply:
> nested classes may not contain nested methods

Ah, yes, _that problem_ again. :smile:

Here's a slightly different idea you can consider if it is acceptable:

Since you're already forced to host the argument factory in the top-level class, you could switch to using `@ArgumentsSource` instead of `@MethodSource`.  This has the following benefits:

* It allows you to move the test methods back inside the nested fixture.
* It removes the need to suppress the unused method warning because the factory is now referenced statically rather than via reflection.

But it has the following drawbacks:

* The factory is no longer in the same scope as the test methods.
* It's a few more lines of code for the class declaration beyond just the method definition.

Reply:
That's fine as it's consistent with most of our data-driven tests.  The addition of `@ParameterizedTest` recently was to overcome the smell of a Mockito `reset()`, which isn't present here.  So, old-school list is good. :+1: